src/test/java/com/sggc/controllers/SggcControllerIT.java

```diff
@@ -0,0 +1,220 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(controllers = SggcController.class)
+public class SggcControllerIT {
+
+    public static final String SGGC_CONTROLLER_URI = "/api/sggc/";
+
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper mapper;
+    @MockBean
+    private UserService userService;
+    @MockBean
+    private VanityUrlService vanityUrlService;
+    @MockBean
+    private GameService gameService;
+
+    @Test
+    @DisplayName("Given a request with less than two Steam id's when validating the request body then a 400 error will be returned with an appropriate message")
+    public void givenRequestWithLessThanTwoSteamIdsWhenServicesValidatesA400ErrorWillBeReturnedWithAnAppropriateMessage() throws Exception {
+        GetCommonGamesRequest sggcRequest = new GetCommonGamesRequest();
+        sggcRequest.setMultiplayerOnly(true);
+        String mockSteamId = "someSteamId";
+        sggcRequest.setSteamIds(Collections.singleton(mockSteamId));
+
+
+        final ApiError error = new ApiError(
+                "Exception",
+                "Request body violates validation rules, check error details for more information.",
+                Collections.singleton("More than one Steam id must be included")
+        );
+        SggcResponse expectedResponse = new SggcResponse(false, error);
+
+        mockMvc.perform(post(SGGC_CONTROLLER_URI)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(mapper.writeValueAsString(sggcRequest)))
+                .andExpect(status().isBadRequest())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.content().json(mapper.writeValueAsString(expectedResponse), false));
+    }
+
+    @Test
+    @DisplayName("Given a request is received with an invalid Vanity URL when validating the Vanity URL then a 400 error will be return with an appropriate response object")
+    public void givenARRequestIsReceivedWithAnInvalidVanityUrlWhenValidatingThenA400ErrorWillBeReturnedWithAnAppropriateResponseObject() throws Exception {
+        GetCommonGamesRequest sggcRequest = new GetCommonGamesRequest();
+        sggcRequest.setMultiplayerOnly(true);
+        String mockVanityUrl = "someVanityUrl!%$%£%$R";
+        String mockSteamId = "someSteamId";
+        sggcRequest.setSteamIds(Set.of(mockVanityUrl,mockSteamId));
+
+        ValidationResult expectedValidationError = new ValidationResult(true, mockVanityUrl, VANITY_URL_NOT_ALPHANUMERIC_ERROR_MESSAGE);
+        when(vanityUrlService.validateSteamIdsAndVanityUrls(Set.of(mockVanityUrl,mockSteamId))).thenReturn(List.of(expectedValidationError));
+
+
+        final ApiError error = new ApiError(
+                "ValidationException",
+                "Request body violates validation rules, check error details for more information.",
```

Comment:
Update messages and tests...
"Request body violates validation rules. Please review the response object for more information."